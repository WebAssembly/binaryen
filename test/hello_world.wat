(module
 (type $0 (func (param i32 i32) (result i32)))
 (memory $0 256 256)
 (export "add" (func $add))
 (func $add (; 0 ;) (type $0) (param $x i32) (param $y i32) (result i32)
  (i32.add
   (local.get $x)
   (local.get $y)
  )
 )
)
