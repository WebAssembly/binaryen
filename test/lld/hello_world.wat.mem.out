(module
 (type $i32_=>_i32 (func (param i32) (result i32)))
 (type $none_=>_i32 (func (result i32)))
 (type $none_=>_none (func))
 (type $i32_=>_none (func (param i32)))
 (type $i32_i32_=>_i32 (func (param i32 i32) (result i32)))
 (import "env" "puts" (func $puts (param i32) (result i32)))
 (memory $0 2)
 (table $0 1 1 funcref)
 (global $global$0 (mut i32) (i32.const 66128))
 (global $global$1 i32 (i32.const 581))
 (export "memory" (memory $0))
 (export "__wasm_call_ctors" (func $__wasm_call_ctors))
 (export "main" (func $main))
 (export "__data_end" (global $global$1))
 (export "stackSave" (func $stackSave))
 (export "stackAlloc" (func $stackAlloc))
 (export "stackRestore" (func $stackRestore))
 (export "__growWasmMemory" (func $__growWasmMemory))
 (func $__wasm_call_ctors (; 1 ;)
  (nop)
 )
 (func $__original_main (; 2 ;) (result i32)
  (drop
   (call $puts
    (i32.const 568)
   )
  )
  (i32.const 0)
 )
 (func $main (; 3 ;) (param $0 i32) (param $1 i32) (result i32)
  (call $__original_main)
 )
 (func $stackSave (; 4 ;) (result i32)
  (global.get $global$0)
 )
 (func $stackAlloc (; 5 ;) (param $0 i32) (result i32)
  (local $1 i32)
  (global.set $global$0
   (local.tee $1
    (i32.and
     (i32.sub
      (global.get $global$0)
      (local.get $0)
     )
     (i32.const -16)
    )
   )
  )
  (local.get $1)
 )
 (func $stackRestore (; 6 ;) (param $0 i32)
  (global.set $global$0
   (local.get $0)
  )
 )
 (func $__growWasmMemory (; 7 ;) (param $newSize i32) (result i32)
  (memory.grow
   (local.get $newSize)
  )
 )
)
(;
--BEGIN METADATA --
{
  "staticBump": 13,
  "tableSize": 1,
  "initializers": [
    "__wasm_call_ctors"
  ],
  "declares": [
    "puts"
  ],
  "externs": [
  ],
  "implementedFunctions": [
    "___wasm_call_ctors",
    "_main",
    "_stackSave",
    "_stackAlloc",
    "_stackRestore",
    "___growWasmMemory"
  ],
  "exports": [
    "__wasm_call_ctors",
    "main",
    "stackSave",
    "stackAlloc",
    "stackRestore",
    "__growWasmMemory"
  ],
  "namedGlobals": {
    "__data_end" : "581"
  },
  "invokeFuncs": [
  ],
  "features": [
  ],
  "mainReadsParams": 0
}
-- END METADATA --
;)
